test/Microsoft.Azure.WebJobs.Host.UnitTests/Executors/FunctionExecutorTests.cs

```diff
@@ -393,6 +432,44 @@ public async Task ExecuteLoggingAsync_WithDrainModeManagerEnabled_RequestsCancel
             }
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task ExecuteWithRetries_Test(bool invocationThrows)
+        {
+            var functionInstanceEx = (IFunctionInstanceEx)CreateFunctionInstance(Guid.NewGuid(), invocationThrows);
+            var logger = new TestLogger("Tests.FunctionExecutor");
+            var functionExecutor = GetTestFunctionExecutor();
+            // Arrange
+            HostStartedMessage testMessage = new HostStartedMessage();
+            functionExecutor.HostOutputMessage = testMessage;
+
+            int maxRetryCount = 5;
+            TimeSpan delay = TimeSpan.FromSeconds(1);
+            var mockRetryStrategy = new Mock<IRetryStrategy>();
+            mockRetryStrategy.Setup(p => p.MaxRetryCount).Returns(maxRetryCount);
+            mockRetryStrategy.Setup(p => p.GetNextDelay(It.IsAny<RetryContext>())).Returns(delay);
+
+            await functionExecutor.ExecuteWithRetriesAsync(functionInstanceEx, CancellationToken.None, logger, mockRetryStrategy.Object);
+            if (invocationThrows)
+            {
+                mockRetryStrategy.Verify(p => p.GetNextDelay(It.IsAny<RetryContext>()), Times.Exactly(maxRetryCount - 1));
+                var messages = logger.GetLogMessages().Select(p => p.FormattedMessage);
+                Assert.Equal(9, messages.Count());
+                var delayMessages = messages.Where(p => p.Contains($"Waiting for {delay}"));
+                Assert.Equal(4, delayMessages.Count());
+                var executeMessages = messages.Where(p => p.Contains("Function execution attempt"));
+                Assert.Equal(5, executeMessages.Count());
+            }
+            else
+            {
+                mockRetryStrategy.Verify(p => p.GetNextDelay(It.IsAny<RetryContext>()), Times.Never);
+                var messages = logger.GetLogMessages().Select(p => p.FormattedMessage);
+                Assert.Single(messages);
+                Assert.Equal($"Function execution attempt: '0'. Max retry count: '{maxRetryCount}'", messages.ElementAt(0));
```

Comment:
As mentioned above, I don't think we should log anything in this case

Reply:
Udpated!